pkg/protocol/block_dispatcher.go

```diff
@@ -0,0 +1,345 @@
+		b.runTask(func() {
+			b.protocol.HandleError(b.processWarpSyncRequest(commitmentID, src))
+		}, b.warpSyncWorkers)
+	})
+
+	b.protocol.Events.Network.WarpSyncResponseReceived.Hook(func(commitmentID iotago.CommitmentID, blockIDs iotago.BlockIDs, merkleProof *merklehasher.Proof[iotago.Identifier], src network.PeerID) {
+		b.runTask(func() {
+			b.protocol.HandleError(b.processWarpSyncResponse(commitmentID, blockIDs, merkleProof, src))
+		}, b.warpSyncWorkers)
+	})
+}
+
+// processWarpSyncRequest processes a WarpSync request.
+func (b *BlockDispatcher) processWarpSyncRequest(commitmentID iotago.CommitmentID, src network.PeerID) error {
+	// TODO: check if the peer is allowed to request the warp sync
+
+	committedSlot, err := b.protocol.MainEngineInstance().CommittedSlot(commitmentID.Index())
+	if err != nil {
+		return ierrors.Wrapf(err, "failed to get slot %d (not committed yet)", commitmentID.Index())
+	}
+
+	commitment, err := committedSlot.Commitment()
+	if err != nil {
+		return ierrors.Wrapf(err, "failed to get commitment from slot %d", commitmentID.Index())
+	} else if commitment.ID() != commitmentID {
+		return ierrors.Wrapf(err, "commitment ID mismatch: %s != %s", commitment.ID(), commitmentID)
+	}
+
+	blockIDs, err := committedSlot.BlockIDs()
+	if err != nil {
+		return ierrors.Wrapf(err, "failed to get block IDs from slot %d", commitmentID.Index())
+	}
+
+	roots, err := committedSlot.Roots()
+	if err != nil {
+		return ierrors.Wrapf(err, "failed to get roots from slot %d", commitmentID.Index())
+	}
+
+	b.protocol.networkProtocol.SendWarpSyncResponse(commitmentID, blockIDs, roots.TangleProof(), src)
+
+	return nil
+}
+
+// processWarpSyncResponse processes a WarpSync response.
+func (b *BlockDispatcher) processWarpSyncResponse(commitmentID iotago.CommitmentID, blockIDs iotago.BlockIDs, merkleProof *merklehasher.Proof[iotago.Identifier], _ network.PeerID) error {
+	if b.processedWarpSyncRequests.Has(commitmentID) {
+		return nil
+	}
+
+	chainCommitment, exists := b.protocol.ChainManager.Commitment(commitmentID)
+	if !exists {
+		return ierrors.Errorf("failed to get chain commitment for %s", commitmentID)
+	}
+
+	targetEngine := b.targetEngine(chainCommitment)
+	if targetEngine == nil {
+		return ierrors.Errorf("failed to get target engine for %s", commitmentID)
+	}
+
+	acceptedBlocks := ads.NewSet[iotago.BlockID](mapdb.NewMapDB(), iotago.BlockID.Bytes, iotago.SlotIdentifierFromBytes)
+	for _, blockID := range blockIDs {
+		_ = acceptedBlocks.Add(blockID) // a mapdb can newer return an error
+	}
+
+	if !iotago.VerifyProof(merkleProof, iotago.Identifier(acceptedBlocks.Root()), chainCommitment.Commitment().RootsID()) {
+		return ierrors.Errorf("failed to verify merkle proof for %s", commitmentID)
+	}
+
+	b.pendingWarpSyncRequests.StopTicker(commitmentID)
+
+	b.processedWarpSyncRequests.Add(commitmentID)
+
+	targetEngine.Events.BlockDAG.BlockSolid.Hook(func(block *blocks.Block) {
+		block.ID()
+	})
```

Comment:
?

Reply:
Oh, this is something that was supposed to track if a slot was fully solidified as we didnt see the engine advance after we fed all the blocks inside, but we stopped to implement this function and then found the problem to be in the engine.

This was simply forgotten to be removed.